nvim

thiswillbeyourgithub opened the pull request:
according to there : # https://old.reddit.com/r/neovim/comments/5con51/where_is_neovims_viminfo_located/

neovim uses something else than .viminfo files, they are inside .local

also I don't know if the --appimage was necessary, it was the case for my setup


If your PR isn't about profiles or you have no idea how to do one of these, skip the following and go ahead with this PR.

If you make a PR for new profiles or changeing profiles please do the following:
 - The ordering of options follow the rules descripted in [/usr/share/doc/firejail/profile.template](https://github.com/netblue30/firejail/blob/master/etc/templates/profile.template).  
   > Hint: The profile-template is very new, if you install firejail with your package-manager, it maybe missing, therefore, and to follow the latest rules, it is recommended to use the template from the repository.
 - Order the arguments of options alphabetical, you can easy do this with the [sort.py](https://github.com/netblue30/firejail/tree/master/contrib/sort.py).  
 The path to it depends on your distro:

   | Distro | Path |
   | ------ | ---- |
   | Arch/Fedora | `/usr/lib64/firejail/sort.py` |
   | Debian/Ubuntu/Mint | `/usr/lib/x86_64-linux-gnu/firejail/sort.py` |
   | local git clone | `contrib/sort.py` |

   Note also that the sort.py script exists only since firejail `0.9.61`.

See also [CONTRIBUTING.md](/CONTRIBUTING.md).

rusty-snake commented on etc/nvim.profile:
```diff
@@ -0,0 +1,26 @@
+# Firejail profile for vim
+# This file is overwritten after every install/update
+# Persistent local customizations
+include /etc/firejail/vim.local
```
```suggestion
include nvim.local
```

rusty-snake commented on etc/nvim.profile:
```diff
@@ -0,0 +1,26 @@
+# Persistent local customizations
+include /etc/firejail/vim.local
+# Persistent global definitions
+include /etc/firejail/globals.local
```
```suggestion
include globals.local
```

rusty-snake commented on etc/nvim.profile:
```diff
@@ -0,0 +1,26 @@
+# Firejail profile for vim
```
```suggestion
# Firejail profile for neovim
```

rusty-snake commented on etc/nvim.profile:
```diff
@@ -0,0 +1,26 @@
+
+include /etc/firejail/disable-common.inc
+include /etc/firejail/disable-passwdmgr.inc
+include /etc/firejail/disable-programs.inc
```
```suggestion
include disable-common.inc
include disable-passwdmgr.inc
include disable-programs.inc
```

rusty-snake commented on etc/nvim.profile:
```diff
@@ -0,0 +1,26 @@
+
+noblacklist ${HOME}/.config/nvim
+noblacklist ${HOME}/.local/share/nvim
+
```
```suggestion

# Allows files commonly used by IDEs
include allow-common-devel.inc

```

rusty-snake commented on etc/nvim.profile:
```diff
@@ -0,0 +1,26 @@
+include /etc/firejail/globals.local
+
+noblacklist ${HOME}/.config/nvim
+noblacklist ${HOME}/.local/share/nvim
```
Add blacklist entry to disable-programs.inc/disable-common.inc

matu3ba commented:
@thiswillbeyourgithub 
Do you manually link your appimages to /usr/local/bin or how do you handle appimage updates?  
Doesnt this mess with `firecfg`?

thiswillbeyourgithub commented:
> @thiswillbeyourgithub
> Do you manually link your appimages to /usr/local/bin or how do you handle appimage updates?
> Doesnt this mess with `firecfg`?

Hi,
I'm sorry but this is one of my first ever pull requests and I have only a basic understanding of all this. when running neovim inside a firejail I was surprised to see that I was apparently using an appimage. I thought I installed it with apt.

To answer your question, I think I moved the appimage there myself and renamed it "nvim". I receive a mail when there is a new nvim release on github.

matu3ba commented:
@thiswillbeyourgithub "Firecfg - Desktop integration utility for Firejail software." comes bundled with firejail on master.  
Okay, you might want to check out [this](https://github.com/AppImage/AppImageUpdate/releases).

However /usr/bin and /usr/local/bin need root access, but I guess I will just invoke a little command script for overwriting the binary.

thiswillbeyourgithub commented:
Thanks for the suggestion, I will keep updating manually though. I think I have to close this now, feel free to reopen if I did anything wrong. Thanks !
